android/library/src/main/java/com/liferay/mobile/screens/ddl/model/NumberField.java

```diff
@@ -113,4 +119,25 @@ private NumberFormat getLabelFormatter() {
         }
         return labelFormatter;
     }
+
+    private boolean checkIsValid() {
+        if (isRequired()) {
+            setValidationState(ValidationState.INVALID_BY_REQUIRED_WITHOUT_VALUE);
+        } else {
+            setValidationState(ValidationState.VALID);
+        }
+
+        return !isRequired();
```

Comment:
This is a bit strange, I would prefer something like:

```
if (isRequired()) {
     setValidationState(ValidationState.INVALID_BY_REQUIRED_WITHOUT_VALUE);
     return false;
} else {
     setValidationState(ValidationState.VALID);
     return true;
}
```
What do you think?

Reply:
Yeah, it's better as you suggested.